### PR TITLE
CON-163: Added a unit test for license error (fix is in core)

### PR DESCRIPTION
**Pull request by samuelraeburn**
@alexcamposruiz The test fails (expected since I haven't fixed it yet). I'm not sure of the actual problem. Do you have any ideas?

### CODE REVIEW

**Comment by samuelraeburn** [comment]
Tests fails as:
`sam@rti-10678:~/working/trees/connector/connector-py$ python3 -m pytest ./test/python/test_rticonnextdds_threading.py
======================================== test session starts =========================================
platform linux -- Python 3.7.3, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: /home/sam/working/trees/connector/connector-py
collected 1 item                                                                                     

test/python/test_rticonnextdds_threading.py F                                                  [100%]

============================================== FAILURES ==============================================
_________________________ TestThreading.test_creation_of_multiple_connectors _________________________

self = <test_rticonnextdds_threading.TestThreading object at 0x7f05acbe4da0>

    def test_creation_of_multiple_connectors(self):
      sem = threading.Semaphore()
      xml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../xml/TestConnector3.xml")
      def outputThread():
          connector = None
          try:
              sem.acquire(True)
              connector = rti.Connector("MyParticipantLibrary::MyPubParticipant", xml_path)
              assert connector is not None
              the_output = connector.getOutput("MyPublisher::MySquareWriter")
              sem.release()
              assert the_output is not None
          finally:
              if connector is not None:
                  connector.delete()
    
      def inputThread():
          connector = None
          try:
              sem.acquire(True)
              connector = rti.Connector("MyParticipantLibrary::MySubParticipant", xml_path)
              assert connector is not None
              the_input = connector.getInput("MySubscriber::MySquareReader")
              sem.release()
              assert the_input is not None
    
          finally:
              if connector is not None:
                  connector.delete()
    
      input_thread = RtiTestThread(inputThread)
      output_thread = RtiTestThread(outputThread)
      input_thread.start()
      output_thread.start()
      input_thread.join(5000)
>     output_thread.join(5000)

test/python/test_rticonnextdds_threading.py:81: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/python/test_rticonnextdds_threading.py:31: in join
    raise self.error
test/python/test_rticonnextdds_threading.py:24: in run
    threading.Thread.run(self)
/usr/local/lib/python3.7/threading.py:865: in run
    self._target(*self._args, **self._kwargs)
test/python/test_rticonnextdds_threading.py:53: in outputThread
    connector = rti.Connector("MyParticipantLibrary::MyPubParticipant", xml_path)
rticonnextdds_connector/rticonnextdds_connector.py:1240: in __init__
    _check_entity_creation(self.native, "Connector")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

entity = None, entity_name = 'Connector'

    def _check_entity_creation(entity, entity_name):
        if entity is None:
            raise Error("Failed to create " + entity_name + ": "
>                       + _get_last_dds_error_message())
E           rticonnextdds_connector.rticonnextdds_connector.Error: Failed to create Connector:

rticonnextdds_connector/rticonnextdds_connector.py:93: Error
====================================== 1 failed in 3.69 seconds ======================================
DDS_PropertySeq_add_element:!new element. ELEMENT 'dds.license.internal_license_string' ALREADY EXISTS. EITHER REMOVE THIS CALL OR CALL assert_element INSTEAD.
DDS_PropertyQosPolicyHelper_add_property:!Add property: dds.license.internal_license_string
RTI Connector Unable to access RTI Data Distribution Service Core Libraries
See www.rti.com for more information.
RTIDDSConnector_newI:!new unable to add license
RTIDDSConnector_EntitiesLookupList_delete:RTIDDSConnector
sam@rti-10678:~/working/trees/connector/connector-py$ 
`

File: test/python/test_rticonnextdds_threading.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+###############################################################################
+# (c) 2005-2015 Copyright, Real-Time Innovations.  All rights reserved.       #
```
Comment (samuelraeburn) [inline test/python/test_rticonnextdds_threading.py:2]: Update date of this copyright

File: test/xml/TestConnector3.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0"?>
```
Comment (samuelraeburn) [inline test/xml/TestConnector3.xml:1]: @alexcamposruiz Had to add this file since the issue only occurs when there is no participant_qos.

Comment (alexcamposruiz) [reply test/xml/TestConnector3.xml:1]: Simplify the file. No need for qos_library, define a dummy type, and only 1 participant.

File: test/python/test_rticonnextdds_threading.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  # In this test we create two Connector objects in separate threads. From one
+  # of the connectors we create an input, from the other an output and check
+  # that communication can occur.
+  # In order to ensure we are testing for CON-163 bug, the XML file does not
+  # contain a <participant_qos> tag.
+  def test_creation_of_multiple_connectors(self):
+    sem = threading.Semaphore()
+    xml_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "../xml/TestConnector3.xml")
+    def outputThread():
```
Comment (alexcamposruiz) [inline test/python/test_rticonnextdds_threading.py:49]:
```suggestion
    def output_thread():
```

File: test/python/test_rticonnextdds_threading.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            connector = rti.Connector("MyParticipantLibrary::MyPubParticipant", xml_path)
+            assert connector is not None
+            the_output = connector.getOutput("MyPublisher::MySquareWriter")
+            sem.release()
+            assert the_output is not None
+        finally:
+            if connector is not None:
+                connector.delete()
+
+    def inputThread():
```
Comment (alexcamposruiz) [inline test/python/test_rticonnextdds_threading.py:62]:
```suggestion
    def input_thread():
```

File: test/python/test_rticonnextdds_threading.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        connector = None
+        try:
+            sem.acquire(True)
+            connector = rti.Connector("MyParticipantLibrary::MySubParticipant", xml_path)
```
Comment (alexcamposruiz) [inline test/python/test_rticonnextdds_threading.py:66]: Should you use `with open_connector`?

File: test/python/test_rticonnextdds_threading.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+import pytest,time,sys,os,ctypes,json
+sys.path.append(os.path.dirname(os.path.realpath(__file__)) + "/../../")
+import rticonnextdds_connector as rti
+from test_utils import send_data, wait_for_data, open_test_connector
+import threading
+
+# In order to be able to catch failures in threads, we need to wrap the
+# threading.Thread class such that it re-raises exceptions
+class RtiTestThread(threading.Thread):
```
Comment (alexcamposruiz) [inline test/python/test_rticonnextdds_threading.py:17]:
```suggestion
class TestThread(threading.Thread):
```

File: test/python/test_rticonnextdds_threading.py (hunk 5 above)
Comment (samuelraeburn) [reply test/python/test_rticonnextdds_threading.py:66]: Also using `with sem:`

**Pull request merged**